lib/Conversion/FIRRTLToRTL/LowerToRTL.cpp

```diff
@@ -1246,28 +1246,43 @@ void FIRRTLLowering::emitRandomizePrologIfNeeded() {
   randomizePrologEmitted = true;
 }
 
-void FIRRTLLowering::initializeRegister(Value reg, Value resetSignal) {
+void FIRRTLLowering::initializeRegister(Value reg, Value resetSignal,
+                                        Value resetValue) {
   // Emit the initializer expression for simulation that fills it with random
   // value.
   builder->create<sv::IfDefOp>("!SYNTHESIS", [&]() {
     builder->create<sv::InitialOp>([&]() {
-      emitRandomizePrologIfNeeded();
+      if (!RegInitFixed)
+        emitRandomizePrologIfNeeded();
       auto type = reg.getType().dyn_cast<rtl::InOutType>().getElementType();
 
-      builder->create<sv::IfDefOp>("RANDOMIZE_REG_INIT", [&]() {
-        if (resetSignal) {
-          auto one = builder->create<rtl::ConstantOp>(APInt(1, 1));
-          auto notResetValue = builder->create<rtl::XorOp>(resetSignal, one);
-          builder->create<sv::IfOp>(notResetValue, [&]() {
+      if (resetSignal) {
```

Comment:
I think this logic would be simpler if you handle the regInitFixed case outside of the resetSignal case, and use an early exit:

if (reginitFixed) {
 if (resetSignal) 
   thing = resetValue
 else 
   thing = zero
  create<BPAssign>(..)
  return;
}